tests/unit/test_echostream.py

```diff
@@ -55,6 +56,66 @@ def test_simple_zmqlet():
         z.send_message(msg)
 
 
+def test_dynamic_routing_zmqlet():
+    args = set_pea_parser().parse_args(
+        [
+            '--host-in',
+            '0.0.0.0',
+            '--host-out',
+            '0.0.0.0',
+            '--socket-in',
+            'PULL_CONNECT',
+            '--socket-out',
+            'DEALER_CONNECT',
+            '--dynamic-routing',
+            'True',
+            '--timeout-ctrl',
+            '-1',
+        ]
+    )
+
+    args2 = set_pea_parser().parse_args(
+        [
+            '--host-in',
+            '0.0.0.0',
+            '--host-out',
+            '0.0.0.0',
+            '--port-in',
+            str(args.port_out),
+            '--port-out',
+            str(args.port_in),
+            '--socket-in',
+            'ROUTER_BIND',
+            '--socket-out',
+            'PUSH_BIND',
+            '--timeout-ctrl',
+            '-1',
+        ]
+    )
+
+    logger = logging.getLogger('zmq-test')
+    with BasePea(args2), Zmqlet(args, logger) as z:
```

Comment:
what is z in thse 2 context managed? why is the Pea needed and how is it validated?